inventory_requests/business_logic/modify_request_cart_logic.py

```diff
@@ -1,21 +1,36 @@
+from datetime import datetime
+
 from inventoryProject.utility.queryset_functions import get_or_not_found
 from items.models import Item
 
 
+def precheck_item(request, modification_type):
+    item = get_or_not_found(Item, pk=request.item.id)
+    return not (item.quantity - request.quantity < 0 and
+                (modification_type == "approved" or modification_type == "disburse"))
+
+
 def can_approve_deny_cancel_disburse_request_cart(request_cart_to_modify, modification_type):
```

Comment:
Refactored the cancel method tho to call a different method which doesn't do the item checks :D